toolchain/check/convert.cpp

```diff
@@ -887,13 +887,38 @@ auto ConvertCallArgs(Context& context, Parse::Node call_parse_node,
           &context.emitter(), [&](auto& builder) {
             CARBON_DIAGNOSTIC(
                 InCallToFunctionSelf, Note,
-                "Initializing self parameter of method declared here.");
-            builder.Note(self_param->parse_node, InCallToFunctionSelf);
+                "Initializing `{0}` parameter of method declared here.",
+                llvm::StringRef);
```

Comment:
```suggestion
                llvm::StringLiteral);
```

Is StringLiteral valid here? StringRef is just something that makes me cautious on lifetime concerns, although this use looks valid.

Reply:
Yes, it's a bit awkward because we can't convert the conditional expression below to `StringLiteral`, but that's not a big problem. Done.